zellij-utils/src/input/theme.rs

```diff
@@ -48,6 +51,65 @@ pub struct PaletteFromYaml {
 pub enum PaletteColorFromYaml {
     Rgb((u8, u8, u8)),
     EightBit(u8),
+    Hex(HexColor),
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct HexColor(u8, u8, u8);
+
+impl From<HexColor> for (u8, u8, u8) {
+    fn from(e: HexColor) -> (u8, u8, u8) {
+        let HexColor(r, g, b) = e;
+        (r, g, b)
+    }
+}
+
+pub struct HexColorVisitor();
+
+impl<'de> Visitor<'de> for HexColorVisitor {
+    type Value = HexColor;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a hex color in the format #RGB or #RRGGBB")
+    }
+
+    fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        if s.len() == 3 {
+            Ok(HexColor(
+                u8::from_str_radix(&s[0..1], 16).map_err(E::custom)? * 0x11,
+                u8::from_str_radix(&s[1..2], 16).map_err(E::custom)? * 0x11,
+                u8::from_str_radix(&s[2..3], 16).map_err(E::custom)? * 0x11,
+            ))
+        } else if s.len() == 6 {
+            Ok(HexColor(
+                u8::from_str_radix(&s[0..2], 16).map_err(E::custom)?,
+                u8::from_str_radix(&s[2..4], 16).map_err(E::custom)?,
+                u8::from_str_radix(&s[4..6], 16).map_err(E::custom)?,
+            ))
+        } else {
+            Err(Error::custom("Hex color must be of form #RGB or #RRGGBB"))
+        }
```

Comment:
It is really cool feature. I like it. But, I think it would be better to express the colors more specifically like '#FF4500'' or ''#3F7''. It seems to make more sense even if the user is presented with an error.

Reply:
I added a commit which now allows the forms:
`ff0000`
`f00`
`"ff0000"`
`"f00"`
`"#ff0000"`
`"#f00"`

The deserializer is passed the string directly, so I can't check for quotes around the hex string.

Reply:
LGTM!